### PR TITLE
fix #2835 address comments

### PR DESCRIPTION
**Pull request by johnpoth**
Addresses comments by @astefanutti in https://github.com/apache/camel-k/pull/2835

Thanks !

**Release Note**
```release-note
NONE
```

### CODE REVIEW

**Review by squakez — APPROVED** [review]

File: pkg/builder/project.go [hunk 1]
```diff
@@ -91,6 +94,16 @@ func generateJavaKeystore(ctx *builderContext) error {
 	return jvm.GenerateKeystore(ctx.C, ctx.Path, ctx.Maven.TrustStoreName, ctx.Maven.TrustStorePass, certsData)
 }
 
+func mergeSecrets(secrets []corev1.SecretKeySelector, secret *corev1.SecretKeySelector) []corev1.SecretKeySelector {
+	if secrets == nil && secret == nil {
```
Comment (tadayosi) [inline pkg/builder/project.go:98]: Is this check necessary?  `secrets == nil`
I think it's always safe to assume a slice is not nil in Go.

File: pkg/trait/openapi.go [hunk 2]
```diff
@@ -272,9 +272,11 @@ func (t *openAPITrait) createNewOpenAPIConfigMap(e *Environment, resource v1.Dat
 		return err
 	}
 	mc.GlobalSettings = data
+	// nolint: staticcheck
+	secrets := mergeSecrets(e.Platform.Status.Build.Maven.CASecrets, e.Platform.Status.Build.Maven.CASecret)
 
-	if e.Platform.Status.Build.Maven.CASecret != nil {
-		certsData, err := kubernetes.GetSecretsRefData(e.Ctx, e.Client, e.Platform.Namespace, e.Platform.Status.Build.Maven.CASecret)
+	if secrets != nil {
```
Comment (tadayosi) [inline pkg/trait/openapi.go:278]: same here

File: pkg/trait/openapi.go [hunk 3]
```diff
@@ -348,6 +350,16 @@ func (t *openAPITrait) createNewOpenAPIConfigMap(e *Environment, resource v1.Dat
 	return nil
 }
 
+func mergeSecrets(secrets []corev1.SecretKeySelector, secret *corev1.SecretKeySelector) []corev1.SecretKeySelector {
+	if secrets == nil && secret == nil {
```
Comment (tadayosi) [inline pkg/trait/openapi.go:354]: same here

File: pkg/util/camel/catalog.go [hunk 4]
```diff
@@ -94,8 +95,10 @@ func GenerateCatalog(
 	}
 
 	var caCerts [][]byte
-	if mvn.CASecret != nil {
-		caCerts, err = kubernetes.GetSecretsRefData(ctx, client, namespace, mvn.CASecret)
+	// nolint: staticcheck
+	secrets := mergeSecrets(mvn.CASecrets, mvn.CASecret)
+	if secrets != nil {
```
Comment (tadayosi) [inline pkg/util/camel/catalog.go:100]: same here

File: pkg/util/camel/catalog.go [hunk 5]
```diff
@@ -104,6 +107,16 @@ func GenerateCatalog(
 	return GenerateCatalogCommon(ctx, globalSettings, []byte(userSettings), caCerts, mvn, runtime, providerDependencies)
 }
 
+func mergeSecrets(secrets []corev1.SecretKeySelector, secret *corev1.SecretKeySelector) []corev1.SecretKeySelector {
+	if secrets == nil && secret == nil {
```
Comment (tadayosi) [inline pkg/util/camel/catalog.go:111]: same here

File: pkg/builder/project.go (hunk 1 above)
Comment (johnpoth) [reply pkg/builder/project.go:98]: Hi @tadayosi ! Thanks for reviewing! 

I didn't know a slice cannot be nil in Go ? 

https://go.dev/tour/moretypes/12#:~:text=The%20zero%20value%20of%20a,and%20has%20no%20underlying%20array.

I just wrote it to be bullet proof but I am happy to remove it if you think it's unnecessary ?

Thanks !

Comment (johnpoth) [reply pkg/builder/project.go:98]: After running it debug mode it seems that `ctx.Build.Maven.CASecrets` does get set to nil so the check is necessary ? Thanks!

Comment (johnpoth) [reply pkg/builder/project.go:98]: Let me merge this for now and we can reiterate later

**Pull request merged**